testutils/src/main/scala/za/co/absa/enceladus/testutils/infoFileComparison/AtumModelUtils.scala

```diff
@@ -0,0 +1,192 @@
+  */
+object ModelDifferenceParser {
+  /**
+    * The method returns JSON representation of a [[ModelDifference]] object
+    */
+  def asJson(modelDifference: List[ModelDifference[_]]): String = {
+    ControlUtils.asJsonPretty[List[ModelDifference[_]]](modelDifference)
+  }
+}
+
+/**
+  * Object holding extensions of Atum Models for comparison purposes.
+  */
+object AtumModelUtils {
+  private val logger: Logger = LogManager.getLogger(this.getClass)
+
+  /**
+    * ControlMeasure's class extension adding compareWith
+    * @param controlMeasure Control Measure instance
+    */
+  implicit class ControlMeasureOps (controlMeasure: ControlMeasure){
+    /**
+      * Compare this ControLMeasure with the one passed in
+      * @param secondControlMeasure Second control measure
+      * @return Returns a list of model differences
+      */
+    def compareWith(secondControlMeasure: ControlMeasure): List[ModelDifference[_]] = {
+      val metadataDifferences = controlMeasure.metadata.compareWith(secondControlMeasure.metadata, "metadata")
+      val checkpointsDifferences = controlMeasure.checkpoints.zipWithIndex.foldLeft(List[ModelDifference[_]]()) {
+        case (agg, (value, index)) =>
+          value.compareWith(secondControlMeasure.checkpoints(index), s"checkpoints[$index]") ::: agg
+      }
+      metadataDifferences ::: checkpointsDifferences
+    }
+  }
+
+  /**
+    * ControlMeasureMetadata's class extension adding compareWith
+    * @param metadata ControlMeasureMetadata instance
+    */
+  implicit class ControlMeasureMetadataOps (metadata: ControlMeasureMetadata){
+    /**
+      * Compare this ControlMeasureMetadata with the one passed in
+      * @param secondMetadata Second control measure metadata
+      * @param curPath Path to the ControlMeasureMetadata through the model
+      * @return Returns a list of model differences
+      */
+    def compareWith(secondMetadata: ControlMeasureMetadata, curPath: String): List[ModelDifference[_]] = {
+      val diffs = simpleCompare(metadata.sourceApplication,
+        secondMetadata.sourceApplication, s"$curPath.sourceApplication") ::
+      simpleCompare(metadata.country, secondMetadata.country, s"$curPath.country") ::
+      simpleCompare(metadata.historyType, secondMetadata.historyType, s"$curPath.historyType") ::
+      simpleCompare(metadata.dataFilename, secondMetadata.dataFilename, s"$curPath.dataFilename") ::
+      simpleCompare(metadata.sourceType, secondMetadata.sourceType, s"$curPath.sourceType") ::
+      simpleCompare(metadata.version, secondMetadata.version, s"$curPath.version") ::
+      simpleCompare(metadata.informationDate, secondMetadata.informationDate, s"$curPath.informationDate") ::
+        Nil
+
+      val additionalInfoDiff = additionalInfoComparison(metadata.additionalInfo,
+        secondMetadata.additionalInfo,
+        s"$curPath.additionalInfo")
+
+      (diffs flatten) ::: additionalInfoDiff
```

Comment:
I'm not a huge fan of postfix notation. And it is [actively discouraged](https://docs.scala-lang.org/style/method-invocation.html):

> This style is unsafe, and should not be used. Since semicolons are optional, the compiler will attempt to treat it as an infix method if it can, potentially taking a term from the next line.
This may result in unexpected compile errors at best, and happily compiled faulty code at worst. Although the syntax is used by some DSLs, it should be considered deprecated, and avoided.


Since scala 2.10 it produces a warning and since [2.13 it won't compile outright](https://github.com/scala/scala/pull/6559) (unless you specify the cpmpiler option `postfixOps` to enable its parsing)

Reply:
@GeorgiChochov Are you sure `:::` is a postfix notation? Isn't it just a binary operator?

Reply:
`:::` is fine, `diffs flatten` is postfix notation (`diffs.flatten()` is preferable).

Reply:
Oh... another reason not to use them. I didn't notice it and saw `diffs_flatten`